docs/half_conversion.rst

```diff
@@ -0,0 +1,51 @@
+half-float Conversion Configuration Options
+###########################################
+
+The Imath library supports three options for conversion between 16-bit
+half and 32-bit float:
+
+1. F16C SSE instructions - single-instruction conversion for machine
+   architectures that support it. When available, this is the fastest
+   option by far.
+
+2. Bit-shift conversion algorithm.
+
+3. Conversion from half to float via a 16-bit lookup table. Prior to
+   Imath v3.1, this was the only method supported.
+
+To use the F16C instruction set on an architecture that supports it,
+simply provide the appropriate compiler flags. For gcc/g++, for example:
+
+    % cmake -dCMAKE_CXX_FLAGS="-m16fc"
```

Comment:
I think we should lay out explicitly at least gcc/clang/msvc. The first two are easy, however, I'm not sure 'simply' is the right word - 

```suggestion
To use the F16C instruction set on an architecture that supports it,
provide the appropriate compiler flags. 

    - gcc/g++/ clang:

    % cmake -dCMAKE_CXX_FLAGS="-m16fc"

    - msvc:
    
    ????
```
@kdt3rd I'm puzzling through the MSVC docs and blogs (e.g. https://walbourn.github.io/directxmath-f16c-and-fma/) Do you know if there's a corresponding /A setting or something for msvc?

Reply:
it's an interesting question. Technically it's an extension to the core SSE functionality on it's own added after AVX (that's how both Intel and AMD talk about it anyway), but MSVC doesn't seem to do that particularly clearly. Per that blog post, they indicate that MSFT will emit it regardless of settings, and it is unclear if they set the __F16C__ #define so maybe it is not as clean as I hoped. When you look in the preprocessor checks in that directx math project, it force F16C on when AVX2 is enabled, so certainly /arch:AVX2 would be safe to mention. But I'm not sure the existing preprocessor checks are sufficient.

Reply:
To clarify, is setting CMAKE_CXX_FLAGS="-m16fc" indeed the proper advice for g++ and clang? 
@kdt3rd, have you been testing on MSVC, and if so, how?

Reply:
In response to Kimball's comment, I searched a bit. I think clang and gcc in addition to having -m16fc (links below) also automatically enable the intrinsics for architectures that have it (gcc link below, search 16fc) like msvc. I bet, but haven't found a reference, that clang does as well.

clang and gcc have the switch. gcc documents that many of the architecture targets enable f16c automatically.

clang: https://clang.llvm.org/docs/ClangCommandLineReference.html#cmdoption-clang-mf16c
gcc: https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html

maybe it's enough for us to summarize the flag, and to note that it's automatically available when targeting processors that have it?

Reply:
Are you saying the intrinsics are enabled by default if available? That's not my experience, I'm building on an Unbuntu x86_64 machine and the __F16C__ cpp symbol is only defined when I include "-m16c" on the command line, for both g++ and clang++.

Reply:
NO, they are not enabled by default, because you don't know what kind of machine you'll be running the code on. But they are enabled by a variety of -mfoo flags (not just -mf16c), if the foo architecture is known to always have f16c capabilities.

Reply:
Sorry, I was responding to Nick's questions around MSVC - Microsoft generates the instruction regardless and doesn't have an explicit switch or seemingly good tests if it's around. I have not been testing this part with windows other than validating that the code is correct using godbolt, but rather, I have been testing the windows filesystem stuff for EXR using the low level win32 api to do the pread/pwrite equivalents